src/pamiq_core/console/cui.py

```diff
@@ -0,0 +1,141 @@
+import argparse
+import cmd
+import json
+from typing import override
+
+import requests
+
+
+class Console(cmd.Cmd):
+    intro = 'Welcome to the AMI system console. "help" lists commands.\n'
+    prompt: str
+
+    def __init__(self, host: str, port: int) -> None:
+        super().__init__()
+        self._host = host
+        self._port = port
+        self._fetch_status()
+
+    @override
+    def onecmd(self, line: str) -> bool:
+        # Check command depend on WebAPI
+        cmd_name, _, _ = self.parseline(line)
+        if cmd_name in ["pause", "p", "resume", "r", "ckpt", "c", "shutdown", "s"]:
+            # Check if WebAPI available.
+            if self._fetch_status() == "offline":
+                print(f'Command "{cmd_name}" not executed. Can\'t connect AMI system.')
+                return False
+        # Perform command
+        return super().onecmd(line)
+
+    @override
+    def do_help(self, arg: str) -> None:
+        """Show all commands and details."""
+        # Store methods with the same docstring
+        # groups = {"Doscstring": ["command", "c"]}
+        groups: dict[str, list[str]] = {}
+        for attr in dir(self):
+            if attr.startswith("do_"):
+                cmd_name = attr[3:]
+                method = getattr(self, attr)
+                doc = method.__doc__ or ""
+                doc = doc
+                if doc not in groups:
+                    groups[doc] = []
+                groups[doc].append(cmd_name)
+        # Define showing order
+        _doc_order = [
+            "Show all commands and details.",
+            "Pause the AMI system.",
+            "Resume the AMI system.",
+            "Save a checkpoint.",
+            "Shutdown the AMI system.",
+            "Exit the console.",
+        ]
+        assert set(_doc_order) == set(groups.keys())
+        # Show details in the format f"{c}/{command} {Docstring}"
+        for doc in _doc_order:
+            cmds = "/".join(sorted(groups[doc], key=len))
+            print(f"{cmds:<11} {doc}")
```

Comment:
これもうね、割とディスプレイするテキストの内容って見やすくするために色々調整したり、いじったりすることが考えられるので、複雑なロジックは避けて、ベタ書きするのがメンテナンスがしやすいと思う。
```suggestion
        print("""\
h/help        Show all commands and details.
p/pause     Pause the system.
r/resume    Resume the system.
save           Save a state to disk.
shutdown  Shutdown the system.
q/quit         Exit the console.
""")
```

Reply:
https://github.com/MLShukai/pamiq-core/pull/210/commits/c79f1673c50df4484c73ecff76e965d09cda750f でベタ書きに変更しといた
今後変更する可能性も考慮して、全てのコマンドがhelpに含まれているかのテストも追加しといた